R/forecast.R

```diff
@@ -460,26 +460,26 @@ is_forecast.default <- function(x, ...) {
 #' @rdname is_forecast
 #' @keywords check-forecasts
 is_forecast.forecast_sample <- function(x, ...) {
-  inherits(x, "forecast_sample")
+  inherits(x, "forecast_sample") && inherits(x, "forecast")
 }
 
 #' @export
 #' @rdname is_forecast
 #' @keywords check-forecasts
 is_forecast.forecast_binary <- function(x, ...) {
-  inherits(x, "forecast_binary")
+  inherits(x, "forecast_binary") && inherits(x, "forecast")
 }
 
 #' @export
 #' @rdname is_forecast
 #' @keywords check-forecasts
 is_forecast.forecast_point <- function(x, ...) {
-  inherits(x, "forecast_point")
+  inherits(x, "forecast_point") && inherits(x, "forecast")
 }
 
 #' @export
 #' @rdname is_forecast
 #' @keywords check-forecasts
 is_forecast.forecast_quantile <- function(x, ...) {
-  inherits(x, "forecast_quantile")
+  inherits(x, "forecast_quantile") && inherits(x, "forecast")
 }
```

Comment:
I'm not completely sure this makes sense to me? I don't expect a class-specific method to check if the object inherits from the method. I don't think it needs to be a generic/method at all.

I would probably just have:

```r
is_forecast <- function(x, ...) {
  inherits(x, "forecast")
}
```

Reply:
The argument here is that sometimes you want to be sure that something is of a specific forecast class (e.g. you'd want to call `is_forecast.forecast_quantile()` directly. 
But maybe I'm misunderstanding your point.

Reply:
Oh, I understand now and I agree these functions are useful. But then the name is incorrect in my opinion.

The dot indicates this is a method for `forecast_quantile` objects. In other words, "if `x` of class `forecast_quantile`, `is_forecast()` dispatches to `is_forecast.forecast_quantile()`". This is not what you are describing here.

What you are describing should be named `is_forecast_quantile()`.

Reply:
This is the same point as I was making in my review. I think we just want is_forecast_specific_thing

Reply:
do we want to export all the `is_forecast_something` functions? I think so, but also we already have so many exports...

Reply:
> This just was just supposed to mean as_forecast(),

No?

Reply:
I think we want to move to having all the forecast_<thing> classes having their own constructors/vaidators etc

Reply:
Yeah that's what I meant: every class should have their own `as_forecast_something()` method, not every class should have their `as_forecast_sample.something()` method

Reply:
Done - we now have a standalone `is_forecast` function for every forecast class